python-package/lightgbm/basic.py

```diff
@@ -650,7 +660,7 @@ def _dump_pandas_categorical(pandas_categorical, file_name=None):
 def _load_pandas_categorical(
     file_name: Optional[Union[str, Path]] = None,
     model_str: Optional[str] = None
-) -> Optional[str]:
+) -> Optional[List[List]]:
```

Comment:
this returns the output of `json.loads()` called on the `pandas: categorical` line of a model text file

https://github.com/microsoft/LightGBM/blob/0b937d1074fecd91243cad7a8bc5b99aaac20996/python-package/lightgbm/basic.py#L674

and that line will contain data that `json.loads()` reads in as a list of lists.